pkg/execute/params.go

```diff
@@ -0,0 +1,115 @@
+package execute
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/mattn/go-shellwords"
+	"github.com/spf13/pflag"
+)
+
+const (
+	cantParseCmd            = "cannot parse command. Please use 'help' to see supported commands"
+	incorrectFilterFlag     = "incorrect use of --filter flag: %s"
+	missingCmdFilterValue   = `incorrect use of --filter flag: an argument is missing. use --filter="value" or --filter value`
+	multipleFilters         = "incorrect use of --filter flag: found more than one filter flag"
+	filterFlagParseErrorMsg = `incorrect use of --filter flag: could not parse flag in %s
+error: %s
+Use --filter="value" or --filter value`
+)
+
+var (
+	clusterNameFlagRegex = regexp.MustCompile(`--cluster-name[=|\s]*(\S*)`)
+)
+
+// OptionalParams contains cmd line arguments for executors
+type OptionalParams struct {
+	CleanCmd     string
+	Filter       string
+	ClusterName  string
+	TokenizedCmd []string
+}
```

Comment:
BTW consider renaming this file to `flags` and renaming this struct with e.g. simply Flags?

```suggestion
type Flags struct {
	CleanCmd     string
	Filter       string
	ClusterName  string
	TokenizedCmd []string
}
```